tests/integration/test_interface.py

```diff
@@ -167,8 +147,11 @@ def test_generate_dataset_and_col_noising(
                 != check_noised.loc[shared_not_missing_idx, col_name].values
             ).any()
             # Check that the amount of noising seems reasonable
-            # NOTE: This threshold is guessed at since it's complicated to know accurately
-            threshold = 0.6 if "unit" in col.name else 0.06
+            threshold = 1 - cell_probability ** len(col.noise_types)
```

Comment:
Do you know how close to this threshold we get?

Reply:
We're pretty far away for some columns. I think it actually ranges from like 30-60%. But this calculated threshold is more like 98% (e.g. for three noise types, 1-0.25^3 = 98.4%).

I'm not sure what's better - a more conservative static number or this "smarter" approach.